code-challenges.js

```diff
@@ -39,26 +60,48 @@ var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 
 var vowelTester1 = "learn"
 // Expected output: 1
-var vowelTester1 = "throw"
+var vowelTester2 = "throw"
 // Expected output: 3
-
-
-
-
+const indexVowel = string =>{
+//construct an array with all vowels
+let vowels = "aeiouAEIOU"
+//loop through the string until the first vowel is found
+for (let i = 0; i < string.length; i++){
+    if(vowels.includes(string.charAt(i))) {
+        return i
+        } 
+    }
+}
+console.log(indexVowel(vowelTester1))
+console.log(indexVowel(vowelTester2))
 
 
 // --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
-
-
-
+const calculator = (num1, op, num2) => {
+//check the nums arguments are numbers
+if (typeof(num1) === "number" && typeof(num2) === "number") {
+//if operation is +, then return num1+num2 to the function
+    if(op === "+") return num1 + num2
+//if operation is -, then return num1-num2 to the function
+    else if(op === "-") return num1 - num2
+//if operation is *, then return num1*num2 to the function
+    else if(op === "*") return num1 * num2
+//if operation is / and num2 is 0, then return Can't divide by 0!
+    else if(op === "/" && num2 ===0) return "Can't divide by 0!"
+//if operation is /, then return num1/num2 to the function
+    else if(op === "/" ) return num1 / num2
+//if operation is not on of +-*/, then return an error message
+    else return "enter a valid mathematical operation (+, -, *, /)"
+    } else return "enter numbers"
+}
```

Comment:
great job taking feedback form last week and adding the pseudo-code through these js questions! it's such a good habit to form and it really shows your thought process. excellent job!